src/redux/actions/experiments/renameMetadataTrack.js

```diff
@@ -42,34 +37,16 @@ const updateMetadataTrack = (
     );
 
     dispatch({
-      type: EXPERIMENTS_METADATA_UPDATE,
+      type: EXPERIMENTS_METADATA_RENAME,
       payload: {
         oldKey: oldMetadataKey,
         newKey: newMetadataKey,
         experimentId,
       },
     });
-
-    experiment.sampleIds.forEach((sampleUuid) => {
-      dispatch({
-        type: SAMPLES_UPDATE,
-        payload: {
-          sampleUuid,
-          sample: { metadata: { [newMetadataKey]: samples[sampleUuid].metadata[oldMetadataKey] } },
-        },
-      });
-
-      dispatch({
-        type: SAMPLES_METADATA_DELETE,
-        payload: {
-          metadataKey: oldMetadataKey,
-          sampleUuid,
-        },
-      });
-    });
```

Comment:
I think the real issue was here, if a user adds a space after the metadata name it would not notice that the name had changed (trailing spaces are removed) so what happened here was the following:

User has a metadata track "Track 1"
User sets new name "Track 1 ", the trailing space is removed so:

`oldMetadataKey = 'Track 1'`
`newMetadataKey = 'Track 1'`

- `SAMPLES_UPDATE`: Sets `sample.metadata['Track 1'] = sample.metadata['Track 1']`
- `SAMPLES_METADATA_DELETE`: Deletes `sample.metadata['Track 1']`

So we ended up with a sample with an undefined value for the renamed metadata track, this would cause crashes when there was any further interaction

Reply:
This is where [the fix](https://github.com/hms-dbmi-cellenics/ui/pull/776/files#r925002940) happens

Reply:
really interesting and obscure bug